examples/framecount.py

```diff
@@ -6,112 +6,37 @@
+import ppb
 
 
-class TestController(object):
-    """
-    A controller that raises a Quit event after the given time.
-    """
-
-    def __init__(self, scene, length):
-        """
-        Pass in the scene and the number of seconds to run the
-        script.
-
-        :param scene: Publisher
-        :param length: number
-        :return: TestController
-        """
-        self.countdown = length
-        # need to keep track of how much time is left.
-        # Alternatively could keep the time to run and accumulate.
-
-        scene.subscribe(Tick, self.tick)
-        # subscribe to Tick events with the scene.
-
-    def tick(self, event):
-        """
-        Advance the countdown and raise a Quit event when complete.
-
-        :param event: Tick
-        :return:
-        """
-        self.countdown += -1 * event.sec
-        # Advance countdown. Use -1 * time to be compatible with Python2
-
-        if self.countdown <= 0:
-            engine.message(Quit())
-            # Raise the quit event.
-
-
-class TestView(object):
+class TestScene(ppb.BaseScene):
     """
     A view that prints a test line in place of rendering frames.
     """
 
-    def __init__(self, scene, target_fps):
-        """
-
-        :param scene: Publisher
-        :param target_fps: number of frames per second to render
-        :return:
-        """
-        self.target_render_time = 1/target_fps
-        # Get amount of time between render calls.
+    duration = 1.0
 
-        self.countdown = self.target_render_time
-        # A countdown to control render speed.
-        self.count = 0
-        # A simple accumulator to count frames.
-        scene.subscribe(Tick, self.tick)
-        # Subscribe to the scene.
+    def __init__(self, *p, **kw):
+        super().__init__(*p, **kw)
 
-    def tick(self, event):
-        """
-        Advance render countdown, render, and reset counter.
+        self.frames = 0
+        self.start_time = time.monotonic()
 
-        :param event: Tick
-        :return:
-        """
-        self.countdown += -1 * event.sec
-        # Advance countdown. Use -1 * time to be compatible with Python2
+    def on_update(self, event, signal):
+        # Fires at the update rate (~60 times a second)
+        t = time.monotonic() - self.start_time
+        if t >= self.duration:
+            signal(ppb.events.Quit())
 
-        if self.countdown <= 0:
-            self.countdown = self.target_render_time
-            # Reset the countdown.
-            self.count += 1
-            # Advance frame count.
-            print("Frame {} rendered at {}".format(self.count, event.run_time))
-            # Render information
+    def on_pre_render(self, event, signal):
+        # Fires each frame, which varies from the update rate
+        # We minimize the amount of work done here to maximize framerate
+        t = time.monotonic() - self.start_time
+        self.frames += 1
+        print(f"Frame {self.frames} rendered at {t}")
```

Comment:
So the docstring in this file mentioned log level, we should probably be using that instead of print here.

Reply:
That's not bad for this line. I would argue with that if we had a summary print at the end.

Reply:
Mostly, I want the opening docstring to match whatever the behavior is.

Reply:
Oh, fair.